### ISSUE TITLE
Issues with python sp_execute_external_script and ntext data

### ISSUE THREAD

**Issue by evotjh** [Microsoft/sql-server-samples#279]
﻿I have stored documents as ntext data and their corresponding labels as int in sql2017.

When I use a stored procedure :

CREATE PROCEDURE generate_patents_py_model (@trained_model varbinary(max) OUTPUT)
AS
BEGIN
    EXECUTE sp_execute_external_script
      @language = N'Python'
    , @script = N'
import pickle
import numpy as np

from sklearn.feature_extraction.text import TfidfTransformer, CountVectorizer
from sklearn.linear_model import SGDClassifier
from sklearn.model_selection import StratifiedShuffleSplit
from sklearn.svm import SVC
from sklearn.pipeline import Pipeline
from sklearn.model_selection import cross_val_score, GridSearchCV

df = patent_data
(*** training code here ***) 

trained_model = pickle.dumps(category_predictor)
'
    , @input_data_1 = N'select * from (SELECT TOP 100 PatText, Klasse FROM LPIPES_fulltext_all order by Klasse ASC UNION ALL SELECT TOP 10 PatText, Klasse FROM LPIPES_fulltext_all ORDER BY Klasse DESC) dum'
    , @input_data_1_name = N'patent_data'
    , @params = N'@trained_model varbinary(max) OUTPUT'
    , @trained_model = @trained_model OUTPUT;
END;
GO

I get the following error:

Input data query returns column #0 of type 'ntext' which is not supported by the runtime for 'Python' script. Unsupported types are binary, varbinary, timestamp, datetime2, datetimeoffset, time, text, ntext, image, hierarchyid, xml, sql_variant and user-defined type.

Does this mean that it is not possible to use text data as training data? Should one then store instead a sklearn CountVectorizer representation of the document in sql?

Do you have any example code that trains on text data?

Thanks for your help!

**Comment by uc-msft** [comment]
Add CAST(PatText as nvarchar(max)) & use it. The old blob types are not currently supported due to the way they are stored & processed in queries. But you can cast them to the newer types in your query & use it.
